core/src/main/java/org/bitcoinj/core/VersionMessage.java

```diff
@@ -98,7 +98,7 @@ public VersionMessage(NetworkParameters params, byte[] payload) throws ProtocolE
     public VersionMessage(NetworkParameters params, int newBestHeight) {
         super(params);
         clientVersion = params.getProtocolVersionNum(NetworkParameters.ProtocolVersion.CURRENT);
-        localServices = 0;
+        localServices = NODE_WITNESS;
```

Comment:
Broadcasting of txns simply sends them as a tx message, unannounced. See the comment in PeerGroup.broadcastTransaction(). But if you're sure announcing the flag is correct I'll of course trust you and merge this soon.